core/src/main/java/com/novoda/noplayer/player/PlayerFactory.java

```diff
@@ -18,78 +19,75 @@
+        this(context, prioritizedPlayerTypes, new ExoPlayerCreator(), new MediaPlayerCreator());
+    }
+
+    PlayerFactory(Context context, PrioritizedPlayerTypes prioritizedPlayerTypes, ExoPlayerCreator exoPlayerCreator, MediaPlayerCreator mediaPlayerCreator) {
         this.context = context;
-        this.prioritisedPlayers = prioritisedPlayers;
+        this.prioritizedPlayerTypes = prioritizedPlayerTypes;
+        this.exoPlayerCreator = exoPlayerCreator;
+        this.mediaPlayerCreator = mediaPlayerCreator;
     }
 
     public Player create() {
-        return create(DrmHandler.NO_DRM);
+        return create(DrmType.NONE, DrmHandler.NO_DRM);
     }
 
-    public Player create(DrmHandler drm) {
-        for (PlayerType playerType : prioritisedPlayers) {
-            if (playerType.supports(drm)) {
-                return createPlayerForType(playerType, drm);
+    public Player create(DrmType drmType, DrmHandler drmHandler) {
+        for (PlayerType player : prioritizedPlayerTypes) {
+            if (player.supports(drmType)) {
+                return createPlayerForType(player, drmType, drmHandler);
             }
         }
-        throw UnableToCreatePlayerException.contentNotSupported();
+        throw UnableToCreatePlayerException.unknownDrmType(drmType);
     }
 
-    private Player createPlayerForType(PlayerType playerType, DrmHandler drm) {
+    private Player createPlayerForType(PlayerType playerType, DrmType drmType, DrmHandler drmHandler) {
         switch (playerType) {
             case MEDIA_PLAYER:
-                return createMediaPlayer();
+                return mediaPlayerCreator.createMediaPlayer(context);
             case EXO_PLAYER:
-                return createExoPlayer(createDrmSessionCreatorFor(drm));
+                DrmSessionCreator drmSessionCreator = createDrmSessionCreatorFor(drmType, drmHandler);
+                return exoPlayerCreator.createExoPlayer(context, drmSessionCreator);
             default:
                 throw UnableToCreatePlayerException.unknownPlayerType(playerType);
         }
     }
 
-    private DrmSessionCreator createDrmSessionCreatorFor(DrmHandler drm) {
-        if (drm == DrmHandler.NO_DRM) {
-            return new NoDrmSessionCreator();
-        } else if (drm instanceof StreamingModularDrm) {
-            ProvisionExecutor provisionExecutor = ProvisionExecutor.newInstance();
-            ProvisioningModularDrmCallback mediaDrmCallback = new ProvisioningModularDrmCallback((StreamingModularDrm) drm, provisionExecutor);
-            return new StreamingDrmSessionCreator(mediaDrmCallback);
-        } else if (drm instanceof DownloadedModularDrm) {
-            return new LocalDrmSessionCreator((DownloadedModularDrm) drm);
-        } else {
-            throw UnableToCreatePlayerException.unknownDrmHandler(drm);
+    private DrmSessionCreator createDrmSessionCreatorFor(DrmType drmType, DrmHandler drmHandler) {
```

Comment:
not yet - the plan is to get rid of this from PlayerFactory anyway (@ouchadam will :grin:) so I didn't want to do code changes around this